Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp

```diff
@@ -52,7 +52,11 @@ void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::Per
     if (cachedResult != WebCore::PermissionState::Prompt)
         return completionHandler(cachedResult);
 
-    m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), WTFMove(completionHandler) });
+    if (descriptor.name == WebCore::PermissionName::Notifications)
+        m_requests.append(PermissionRequest { WTFMove(origin), WTFMove(descriptor), WTFMove(completionHandler), WeakPtr { context } });
```

Comment:
If we're going to use the ScriptExecutionContext as a weak pointer, then it is best to declare a WeakPtr<ScriptExecutionContext>&& argument, rather than a ScriptExecutionContext* argument. Declaring what we need up front can make callers more efficient, and it also makes object lifetime more explicit.

Reply:
This makes sense--the implementation for this pull request has now changed, but I'll keep this idea in mind for the future.